core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -302,19 +323,29 @@ public Void call() throws IOException {
                 segmentPropertiesFetcher.getSegmentPropertiesFromDataMap(dataMapList.get(0));
             Segment segment = segmentDataMapGroup.getSegment();
             if (filter.isResolvedOnSegment(segmentProperties)) {
+              FilterExecuter filterExecuter = FilterUtil
+                  .getFilterExecuterTree(filter.getResolver(), segmentProperties,
+                      null, table.getMinMaxCacheColumns(segmentProperties),
+                      false);
               for (int i = segmentDataMapGroup.getFromIndex();
                    i <= segmentDataMapGroup.getToIndex(); i++) {
                 List<Blocklet> dmPruneBlocklets = dataMapList.get(i).prune(
-                    filter.getResolver(), segmentProperties, partitions);
+                    filter.getResolver(), segmentProperties, partitions, filterExecuter);
                 pruneBlocklets.addAll(addSegmentId(
                     blockletDetailsFetcher.getExtendedBlocklets(dmPruneBlocklets, segment),
                     segment));
               }
             } else {
+              FilterExecuter filterExecuter = FilterUtil
+                  .getFilterExecuterTree(new DataMapFilter(segmentProperties, table,
+                          filter.getNewCopyOfExpression()).getResolver(), segmentProperties,
+                      null, table.getMinMaxCacheColumns(segmentProperties),
+                      false);
               for (int i = segmentDataMapGroup.getFromIndex();
                    i <= segmentDataMapGroup.getToIndex(); i++) {
                 List<Blocklet> dmPruneBlocklets = dataMapList.get(i).prune(
-                    filter.getNewCopyOfExpression(), segmentProperties, partitions, table);
+                    filter.getNewCopyOfExpression(), segmentProperties, partitions, table,
```

Comment:
Avoid deserializing ` filter.getNewCopyOfExpression()` again, as already done in line:341, can reuse the same